conferences/ticketing_storage.go

```diff
@@ -0,0 +1,605 @@
+package conferences
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"encore.dev/storage/sqldb"
+)
+
+// SQLStorage provides a Postgres Flavored storage backend to store ticketing information.
+type SQLStorage struct {
+	tx *sqldb.Tx
+}
+
+const (
+	ticketIDUniqueConstraint = "ticket_id_is_unique"
+	tableSlotClaims          = "slot_claim"
+)
+
+func query(ctx context.Context, tx *sqldb.Tx, statement string, args ...interface{}) (*sqldb.Rows, error) {
+	if tx != nil {
+		return sqldb.QueryTx(tx, ctx, statement, args...)
+	}
+	return sqldb.Query(ctx, statement, args...)
+}
+
+func queryRow(ctx context.Context, tx *sqldb.Tx, statement string, args ...interface{}) *sqldb.Row {
+	if tx != nil {
+		return sqldb.QueryRowTx(tx, ctx, statement, args...)
+	}
+	return sqldb.QueryRow(ctx, statement, args...)
+}
+
+func exec(ctx context.Context, tx *sqldb.Tx, statement string, args ...interface{}) (sql.Result, error) {
+	if tx != nil {
+		return sqldb.ExecTx(tx, ctx, statement, args...)
+	}
+	return sqldb.Exec(ctx, statement, args...)
+}
```

Comment:
the repetition of checking if we are or not inside a transaction would add a lot of unnecessary verbosity I think this quick abstraction, being pretty much a passthrough, makes the calls a bit more readable. As with all though, I am willing to be convinced otherwise.

Reply:
In cases where we have no transaction it's adding mental overhead every time you have to make a call to the database. It is harder to understand for someone new to the code when you have a call like this:
```
row := queryRow(ctx, nil, `SELECT * FROM conference`)
```
Than a call like this:
```
row := sqldb.QueryRow(ctx, `SELECT * FROM conference`)
```

It may be easier to use, but I would push back on it being easier to read, and I think we should optimize for being easy to read.